Time Domain Analysis Functions.R

```diff
@@ -0,0 +1,433 @@
+    
+  }
+  
+} else if (input = "MS") {
+  
+  time_vector <- data[ ,1]
+  
+}
+  
+  minutes <- time_vector / 60000
+  
+# then goes into determining if it is an SDRR or SDARR
+  
+  if (tail(minutes) < 5) {
+    
+    # use SDNN
+    
+    if (input == "RR") {
+    
+      sdRR <- sd(data[,1])
+    
+      return(sdRR)
+  
+    } else if (input == "MS") {
+    
+      diffRR <- diff(data[,1])
+    
+      sddiffRR <- sd(diffRR)
+    
+      return(sddiffRR)
+      
+    }
+
+  } else if (tail(minutes) >= 5) {
+    
+    # use SDARR
+  
+      if (input == "RR") {
+    
+        sdRR <- sd(data[,1])
+    
+        return(sdRR)
+  
+    } else if (input == "MS") {
+    
+        diffRR <- diff(data[,1])
+    
+        sddiffRR <- sd(diffRR)
+    
+        return(sddiffRR)
+  
+        }
+
+    }
+
+  }
+
+
+# pNN50 Values
+
+# tells us how many RR intervals are larger than 50 
+
+# used more in medical rather than psychological research 
+
+
+pNN50_Function <- function(x, input) {
+
+  if (input = "RR") {
+
+    RR <- data[ , 1]
+
+  }
+
+  if (input = "MS") {
+    
+    RR <- diff(data[ , 1])
+    
+  }
+  
+  counter <- 0
+  
+  for (i in 1:length(RR)) {
+    
+    if(RR > 50) {
+      
+      counter <- counter + 1
+      
+    }
+  
+  }
+  
+}
+
+
+
+
+# Descriptive Heart Rate -- MIN + MAX + AVERAGE
+
+Descriptive_HR_Function <- function (data, input) {
+  
+  if (input = "MS") {
+    
+    RR <- diff(data [ ,1])
+    
+  }
+  
+  else if (input = "RR") {
+    
+    RR <- data[ ,1]
+    
+  }
+  
+  return(mean(RR))
+  
+  return(min(RR))
+  
+  return(max(RR))
+  
+}
+
+# Time Point Analysis 
+
+# Not part of the time-domain analysis protocol, but sometimes useful for researchers if they are interested
+
+# in comaring heart rate at a baseline vs. at a particular time-point in the experiment 
+
+Time_Point_Analysis_Function <- function (data, input, time1, time2, time3, time4) {
+  
+  if (input = "MS") {
+    
+    RR <- diff(data [ ,1])
+    
+  }
+  
+  else if (input = "RR") {
+    
+    RR <- data[ ,1]
+    
+  }
+  
+  # make a time variable to be able to referece RR values
+  
+  for (value in 1:nrow(data)) {
+    
+    newsum <- sum(data[1:value, ])
+    
+    time_vector <- append(sumVector, newsum)
+    
+  }
+  
+  seconds_vector <- time_vector / 1000
+  
+  newdata <- cbind(seconds_vector, RR)
+  
+  if (time1 = TRUE) {
+    
+    time1 <- newdata[time1, ]
+    
+  }
+  
+  if (time2 = TRUE) {
+    
+    time2 <- newdata[time2, ]
+    
+  }
+  
+  if (time3 = TRUE) {
+    
+    time3 <- newdata[time3, ]
+    
+  }
+  
+  if (time4 = TRUE) {
+    
+    time4 <- newdata[time4, ]
+    
+  }
+  
+}
+
+
+
+
+# SDARR Index 
+
+# this is where the user can decide if they want multiple analysis completed
+
+# on different segments of the data
+
+# i.e., they have a 8 minute segment, and they want RMSSD values for 
+
+# minute 2, 3, 4, 5
+
+
+Segmenting_Function <- function(data, segment1, segment2, segment3, segment4, segment5) {
+
```

Comment:
Might be more efficient to ask user for the first and last segment, and also retrieve info from all segments in between (= less input arguments).